File: test/test_for_each.cpp
#include <stdio.h>
#include "TRTCContext.h"
#include "DVVector.h"
#include "for_each.h"

int main()
{
	Functor printf_functor = {{}, { "x" }, "        printf(\"%d\\n\", x);\n" };

	int hvec[5] = { 1, 2, 3, 1, 2 };
	DVVector vec("int32_t", 5, hvec);
	TRTC_For_Each(vec, printf_functor);

	return 0;
}

File: CSharp/test/test_for_each/test_for_each.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ThrustRTCSharp;

namespace test_for_each
{
    class test_for_each
    {
        static void Main(string[] args)
        {
            Functor printf_functor = new Functor(new string[]{ "x" }, "        printf(\"%d\\n\", x);\n");
            DVVector vec = new DVVector(new int[] { 1, 2, 3, 1, 2 });
            TRTC.For_Each(vec, printf_functor);
        }
    }
}
